web/models/hn_subscription.ex

```diff
@@ -0,0 +1,68 @@
+defmodule Feedya.HNSubscription do
+  use Feedya.Web, :model
+
+  alias Ecto.Changeset
+  alias Feedya.{Repo, HNSubscription, HNStory}
+
+  @required [:terms, :name, :user_id]
+
+  schema "hn_subscriptions" do
+    field :name, :string
+    field :terms, {:array, :string}
+    belongs_to :user, Feedya.User
+    many_to_many :stories, Feedya.HNStory, join_through: "hn_subscription_stories", on_replace: :delete
+
+    timestamps()
+  end
+
+  @doc """
+  Builds a changeset based on the `struct` and `params`.
+  """
+  def changeset(struct, params \\ %{}) do
+    struct
+    |> cast(params, @required)
+    |> validate_required(@required)
+  end
+
+  def by_user(user) do
+    query = from p in HNSubscription,
+            where: p.user_id == ^user.id,
+            select: p,
+            preload: [:stories]
+    Repo.all(query)
```

Comment:
Should use one interface for query-functions. See https://blog.drewolson.org/composable-queries-ecto/